util/typeutil/stringInt.go

```diff
@@ -0,0 +1,32 @@
+package typeutil
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+type StringInt int
+
+func (st *StringInt) UnmarshalJSON(b []byte) error {
+	//convert the bytes into an interface as this will help us check the type of our value
+	var item interface{}
+	if err := json.Unmarshal(b, &item); err != nil {
+		return err
+	}
+
+	switch v := item.(type) {
+	case int:
+		*st = StringInt(v)
+	case float64:
+		*st = StringInt(int(v))
+	case string:
+		///here convert the string into an int
+		i, err := strconv.Atoi(v)
+		if err != nil {
+			return err
+
+		}
+		*st = StringInt(i)
```

Comment:
should add `default` case to return error if non of the above case match

Reply:
There's a `return nil` at the end of the method to provide the same effect. I suppose it's just a matter of preference.

Reply:
This is not applicable now since we have changed the implementation of the Unmarshal function based on benchmarking results.